dev_guide/cicd.adoc

```diff
@@ -0,0 +1,254 @@
+baked-in set of plugins and configuration that greatly ease the integration between Jenkins and OpenShift combined with the burgeoning list of `*CI/CD*` centric features in
+OpenShift itself, including the Jenkins Pipeline `*BuildConfigStrategy*`, a powerful `*CI/CD*` solution emerges.  That solution includes the ability to auto-provision a Jenkins server to run within
+your OpenShift environment, as well as directly reference and initiate Jenkins Pipeline Jobs.
+
+[[inventory-summary]]
+== Summary of Application Components And Relevant Tools
+
+Spoiler alert:  the "application" is more than just the source code written in say Java, Perl, Python, Ruby, or NodeJS.  It is more now than the static web
+content, the integration scripts, the assoicated configuration for the language specific runtimes for the application, or the application specific archives
+consumed by those language specific runtimes.
+
+By pulling in OpenShift Origin, and its combined foundation of Kubernetes and Docker, additional application artfacts now minimally include:
+- Images with their rich set of metadata and associated tooling
+- Environment Variables and a powerful array of API Objects that are either injected into Containers and Pods for application use, or dictate how the Orchestration Engine (OpenShift on Kubernetes) manages
+those Containers and Pods
+
+And the second bullet there should serve as a reminder:  the lines between development and operations have blurred, and we want to improve the interaction between the two .... we now have "DevOps".
+
+But while the inventory of concerns has increased, many of the concepts and semantics around managing those concerns are thankfully very similar:
+- versioning
+- portability
+- isolation
+- upgrade and update
+- hooks (into various flows)
+
+We'll next map these concerns to what is either available in or accessible from OpenShift to this problem domain.
+
+[[versioning]]
+== Versioning
+
+Versioning in this topic's context refers to designating each of the components of your application to the stage in the `*CI/CD*` pipeline their readiness dictates they should reside.
+
+For your source code (in the classic sense), the OpenShift `*Build*`, `*BuildRequest*`, and `*BuildConfig*` object are of course the starting point.  They expose the most typically used
+features of the `*SCM*` (most notably Git) to access the appropriate version of your source code (commit IDs, branches, context directories, etc.) and build the source code.
+
+But source code management (`*SCM*`) tooling is also leveraged througout the industry for managing many artifacts beyond what classically is defined as source code.
+OpenShift also promotes such a strategy.
+
+And where a `*SCM*` is not appropriate, most notably Images, we have the OpenShift Internal Image Registry and `*ImageStreams*`. 
```

Comment:
s/where a SCM/where SCM/
s/notably Images/notably for managing the evolution of container Images/